### PR TITLE
feat(env-check): add check for vscode insiders

### PR DESCRIPTION
**Pull request by cianmSAP**
#860
https://github.com/SAP/open-ux-tools/issues/805

Provide the option to use the VSCode API and pass the installed extensions to `getEnvironment()`
Check `process.env` for indications that environment-check is being ran in VSCode Insiders, and return the correct extensions installed.

### CODE REVIEW

**Review by Klaus-Keller — COMMENTED** [review]
@cianmSAP, any chance we could incorporate proposal from ticket https://github.com/SAP/open-ux-tools/issues/805 to solve the issue.

**Review by Klaus-Keller — APPROVED** [review]
- Code changes look good
- I've build and tested the code locally using CLI
- Test coverage is great

Thanks @cianmSAP, approved from my side.

File: packages/environment-check/src/checks/get-installed.ts [hunk 1]
```diff
@@ -4,10 +4,12 @@ import { join } from 'path';
 import { spawnCommand, npmCommand } from '../command';
 import { Extensions, NpmModules } from '../types';
 import type { ILogger } from '../types';
+import type { Extension } from 'vscode';
 import { isAppStudio } from '@sap-ux/btp-utils';
 
 const pluginsDirBAS = join('/extbin/plugins');
 const globalNodeModulesBAS = join('/extbin/npm/globals/lib/node_modules');
+const globalNodeModulePnpmBAS = join('/extbin/globals/pnpm/5/node_modules');
```
Comment (IainSAP) [inline packages/environment-check/src/checks/get-installed.ts:12]: Can we please determine this path programmatically. This is specific to a pnpm version and the global loc can be changed by config.

**Review by IainSAP — COMMENTED** [review]
See comments

**Review by IainSAP — APPROVED** [review]
Thanks @cianmSAP

**Pull request merged**